nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment-api/src/main/java/org/nuxeo/ecm/platform/comment/api/CommentImpl.java

```diff
@@ -20,136 +20,243 @@
+import java.util.Arrays;
+import java.util.Calendar;
 import java.util.Collection;
 import java.util.HashSet;
-import java.util.Objects;
 
 import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.impl.SimpleDocumentModel;
 
 /**
  * @since 10.3
  */
 public class CommentImpl implements Comment, ExternalEntity {
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String id;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String parentId;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected Collection<String> ancestorIds = new HashSet<>();
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String author;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String text;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected Instant creationDate;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected Instant modificationDate;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String entityId;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String origin;
 
+    /**
+     * @deprecated since 11.1, not used due to {@link #docModel} usage
+     */
+    @Deprecated(since = "11.1")
     protected String entity;
 
+    /**
+     * {@link DocumentModel} storing the {@link Comment} metadata.
+     * 
+     * @since 11.1
+     */
+    protected DocumentModel docModel;
+
+    /**
+     * @since 11.1
+     */
+    public CommentImpl() {
+        this(SimpleDocumentModel.ofType(COMMENT_DOC_TYPE));
+    }
+
+    /**
+     * Constructor for the document adapter factory.
+     *
+     * @since 11.1
+     */
+    protected CommentImpl(DocumentModel docModel) {
+        this.docModel = docModel;
+        this.docModel.detach(true);
+    }
+
     @Override
     public String getId() {
-        return id;
+        try {
+            return docModel.getId();
+        } catch (UnsupportedOperationException e) {
+            // don't fail when docModel is SimpleDocumentModel
+            return null;
+        }
     }
 
     @Override
+    @Deprecated(since = "11.1")
     public void setId(String id) {
-        this.id = id;
+        // not used
```

Comment:
i am wondering in the case of backward compatibility where a customer call `new CommentImpl ().setId("anyId")`  and later in his code `c.getId()` will be `null` (java comment on` #getId` -> `// don't fail when docModel is SimpleDocumentModel`)

Reply:
I think about it too, except from a test I don't see any valid usage, that's why I didn't provide backward.